src/lib/Scenes/Artwork/Components/ArtworkActions.tsx

```diff
@@ -1,49 +1,98 @@
-import { Box, Flex, Sans } from "@artsy/palette"
+import { Box, color, Flex, Sans } from "@artsy/palette"
 import { ArtworkActions_artwork } from "__generated__/ArtworkActions_artwork.graphql"
+import { ArtworkActionsSaveMutation } from "__generated__/ArtworkActionsSaveMutation.graphql"
+import { Pin } from "lib/Icons/Pin"
 import SearchIcon from "lib/Icons/SearchIcon"
+import { isNull } from "lodash"
 import React from "react"
-import { NativeModules, View } from "react-native"
-import { createFragmentContainer, graphql } from "react-relay"
+import { NativeModules, TouchableWithoutFeedback, View } from "react-native"
+import { commitMutation, createFragmentContainer, graphql, RelayProp } from "react-relay"
 import styled from "styled-components/native"
 
 const Constants = NativeModules.ARCocoaConstantsModule
 
 interface ArtworkActionsProps {
   artwork: ArtworkActions_artwork
+  relay?: RelayProp
 }
 
-export const ArtworkActions: React.FC<ArtworkActionsProps> = () => {
-  return (
-    <View>
-      <Flex flexDirection="row">
-        <UtilButton pr={3}>
-          <Box mr={0.5}>
-            <SearchIcon />
-          </Box>
-          <Sans size="3">Save</Sans>
-        </UtilButton>
-        {Constants.AREnabled && (
-          <UtilButton pr={3}>
+export class ArtworkActions extends React.Component<ArtworkActionsProps> {
+  state = {
+    is_saved: null,
+    isHovered: false,
+  }
+
+  get isSaved() {
+    const isSaved = isNull(this.state.is_saved) ? this.props.artwork.is_saved : this.state.is_saved
+
+    return isSaved
+  }
+
+  handleArtworkSave = () => {
+    const { artwork, relay } = this.props
+    commitMutation<ArtworkActionsSaveMutation>(relay.environment, {
+      mutation: graphql`
+        mutation ArtworkActionsSaveMutation($input: SaveArtworkInput!) {
+          saveArtwork(input: $input) {
+            artwork {
+              id
+              is_saved
+            }
+          }
+        }
+      `,
+      variables: { input: { artwork_id: artwork.id, remove: this.isSaved } },
```

Comment:
I was getting 'unknown artwork' errors when testing this in emission, so I tried changing the `artwork_id` arg to `artwork.internalID` and it worked. I feel a bit lost w/r/t these ID changes, is `internalID` what we should be using for queries that go to gravity now?

Reply:
Yeah, good catch!

And I believe yeah, that's the change. It'll be either `internalID` or `gravityID`, depending (not sure how this is determined exactly since artworks come from gravity but are still `internalID`).